### PR TITLE
cmd/snap,strutil: move lineWrap to WordWrapPadded

### PR DESCRIPTION
**Pull request by Meulengracht**
Moved to strutil as its needed for snapctl model as well, also added unit tests for its none exists. I know there might be some overlap between the tests in WordWrapPadded and WordWrap, but because of the modifications happening to the text before it's handed to WordWrap, it makes sense to make sure we test the same width/padding patterns. It actually revealed a flaw with the width check in WordWrap, which is now done differently, as it did not catch all cases before.

Can this skip spread test?

### CODE REVIEW

File: strutil/strutil.go [hunk 1]
```diff
@@ -300,13 +300,12 @@ func WordWrap(out io.Writer, text []rune, indent, indent2 string, termWidth int)
 	//       However, before this we'd count bytes instead of runes, so we'd be
 	//       even more broken. Think of it as successive approximations... at least
 	//       with this work we share tabwriter's opinion on the width of things!
-	if termWidth < 1 {
```
Comment (bboozzoo) [inline strutil/strutil.go:303]: probably worked only because termwidth defaulted to 40 if size was smaller than that

File: strutil/strutil.go [hunk 2]
```diff
@@ -333,3 +332,35 @@ func WordWrap(out io.Writer, text []rune, indent, indent2 string, termWidth int)
...
 	return err
 }
+
+// runesTrimRightSpace returns text, with any trailing whitespace dropped.
+func runesTrimRightSpace(text []rune) []rune {
+	j := len(text)
+	for j > 0 && unicode.IsSpace(text[j-1]) {
+		j--
+	}
+	return text[:j]
```
[the remark this answers is not in the document]
Comment (Meulengracht) [reply strutil/strutil.go:342]: Let me test this!

File: strutil/strutil.go (hunk 1 above)
Comment (Meulengracht) [reply strutil/strutil.go:303]: it seems if padding was larger than termwidth then it got an out of bounds exception, which was not caught by this check

**Review by stolowski — APPROVED** [review]
LGTM, thanks

**Comment by stolowski** [comment]
> Can this skip spread test?

No, it's good to run the tests in this case; we have a bunch of checks that MATCH on snap command outputs and while this change should not affect anything it's nice to have extra safety check.

File: strutil/strutil.go [hunk 3]
```diff
@@ -333,21 +333,12 @@ func WordWrap(out io.Writer, text []rune, indent, indent2 string, termWidth int)
...
-	return text[:j]
-}
-
 // WordWrapPadded wraps the given text, assumed to be part of a block-style yaml
 // string, to fit into termWidth, preserving the line's indent, and
 // writes it out prepending padding to each line.
 func WordWrapPadded(out io.Writer, text []rune, pad string, termWidth int) error {
 	// discard any trailing whitespace
-	text = runesTrimRightSpace(text)
+	text = []rune(strings.TrimRightFunc(string(text), unicode.IsSpace))
```
Comment (mvo5) [inline strutil/strutil.go:341]: nice catch!

**Review by mvo5 — APPROVED** [review]
Thank you

File: strutil/strutil.go (hunk 3 above)
Comment (Meulengracht) [reply strutil/strutil.go:341]: credits to @bboozzoo !

**Review by bboozzoo — APPROVED** [review]
LGTM
